pennylane/fourier/visualization.py

```diff
@@ -0,0 +1,690 @@
+
+        frequency_range = list(range(n_freqs)) + list(range(-n_freqs + 1, 0))
+
+        for coeff_1, coeff_2 in product(list(range(coeffs.shape[1])), list(range(coeffs.shape[2]))):
+            ax[coeff_1, coeff_2].scatter(
+                coeffs[:, coeff_1, coeff_2].real,
+                coeffs[:, coeff_1, coeff_2].imag,
+                facecolor="white",
+                edgecolor=colour,
+            )
+            ax[coeff_1, coeff_2].set_title(
+                f"{frequency_range[coeff_1]}, {frequency_range[coeff_2]}", fontsize=14
+            )
+            ax[coeff_1, coeff_2].grid(True)
+            ax[coeff_1, coeff_2].set_aspect("equal")
+
+    return ax
+
+
+def plot_1D_from_coeffs(coeffs, ax=None):
+    """Visualize a real-valued 1D periodic function given by a set of Fourier coefficients.
+
+    Args:
+        coeffs (array[complex]): Fourier coefficients of a 1-dimensional
+            function. The array should have length ``2d + 1`` where ``d`` is the
+            degree, i.e., the maximum frequency of present in the coefficients.
+            Such an array may be the output of the numpy/scipy ``fft`` function, or
+            :func:`~.pennylane.fourier.coefficients`.
+        ax (matplotlib.axes.Axes): Axis on which to plot. If None, the
+            current axis from ``plt.gca()`` will be used.
+
+    Returns:
+        matplotlib.axes.Axes: The axis after plotting is complete.
+
+    **Example**
+
+    .. code-block:: python
+
+        import matplotlib as plt
+        from pennylane.fourier import coefficients, plot_1D_from_coeffs
+
+        f = ... # A function in 1 variable
+        n_inputs = 1
+        degree = ... # Degree to which coefficients should be calculated
+
+        # Calculate the Fourier coefficients
+        coeffs = coefficients(f, n_inputs, degree)
+
+        # It is not necessary to create subplots; the current axis will be used here
+        plot_1D_from_coeffs(coeffs)
+
+    """
+    coeffs = _validate_coefficients(coeffs, 1, False)
+
+    ax = ax or plt.gca()
+
+    def reconstructed_function(x):
+        # At each point x, the value of the function is given by
+        # c_0 + c_1 e^{ix} + c_2 e^{2ix} + ... + c_{-n} e^{-inx} + ... c_{-1} e^{-ix}
+
+        n_freqs = len(coeffs) // 2 + (len(coeffs) % 2)
+
+        function_value = 0j
+
+        for freq in range(n_freqs):
+            # Ignore tiny coefficients
+            if np.isclose(coeffs[freq], 0):
+                continue
+
+            if freq == 0:
+                function_value += coeffs[freq]
+            else:
+                function_value += coeffs[freq] * np.exp(freq * 1j * x)
+                function_value += np.conj(coeffs[freq]) * np.exp(-freq * 1j * x)
+
+        return function_value.real
+
+    n_points = 500
+    grid_range = np.linspace(-np.pi, np.pi, n_points)
+
+    ax.plot(grid_range, reconstructed_function(grid_range))
+
+    return ax
+
+
+def plot_2D_from_coeffs(coeffs, ax=None):
```

Comment:
Looking at these again, I think I might remove the `plot_1D_from_coeffs` and `plot_2D_from_coeffs`. I think they are not so unique and relevant for the module, and are easily implemented by a user.